### PR TITLE
[[ ObjectGC ]] More efficient object deletion (7.0)

### PR DESCRIPTION
**Pull request by runrevmark**
The engine will now delete all objects it can after each statement execution.

When an object is created, it is tied to a deletion pool owned by the current
wait loop. If the object is subsequently deleted (through script) whilst that
loop is still running, it will be destroyed after the current command finishes
executing.

If an object survives the current wait loop and subsequently deleted it will
be placed in the most suitable wait loop which still exists, and get deleted
when that is returned to.

### CODE REVIEW

File: docs/notes/bugfix-6791.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+The engine will now flush any recently deleted objects after each command as long as they were
+created during the current event handling loop.
+
+If an object is created during one event handling loop, and then deleted during another nested
+event handling loop it won't be flushed until control returns to the original event handling
+loop.
+
+The upshot is that in tight loops, creating and deleting objects will result in objects being
+flushed immediately, reducing memory usage and making it easier to right object processing
```
Comment (livecodestephen) [inline docs/notes/bugfix-6791.md]: Typo: right -> write

File: engine/src/object.h [hunk 2]
```diff
@@ -807,6 +820,7 @@ class MCObject : public MCDLlist
         return m_is_parent_script;
     }
     
+<<<<<<< HEAD
```
Comment (livecodestephen) [inline engine/src/object.h]: Merge markers!

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 6860004

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 6860004e28bb58e26ad76c18faadcd6eca2b627e

**Comment by livecode-vulcan** [comment]
😎 test success 6860004e28bb58e26ad76c18faadcd6eca2b627e
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/430)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/490)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/376)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/441)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/376)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/367)

**Pull request merged**